packages/wp-story-editor/src/api/story.js

```diff
@@ -38,16 +38,17 @@ const transformGetStoryResponse = (post) => {
     name: embedded?.author?.[0].name || '',
   };
 
-  post.capabilities = {
-    hasPublishAction: Object.prototype.hasOwnProperty.call(
-      links,
-      'wp:action-publish'
-    ),
-    hasAssignAuthorAction: Object.prototype.hasOwnProperty.call(
-      links,
-      'wp:action-assign-author'
-    ),
-  };
+  post.capabilities = {};
+
+  for (const link of Object.keys(links)) {
+    if (!link.startsWith('wp:action-')) {
+      continue;
+    }
+
+    // Turn 'wp:action-assign-author' into 'assign-author'
+    const capability = link.replace('wp:action-', '');
```

Comment:
Could we use [snakeCaseToCamelCase](https://github.com/google/web-stories-wp/blob/8d72b8d75d431fbd4635bc6cfc59e6d7dda69922/packages/wp-story-editor/src/api/utils/snakeCaseToCamelCase.js#L23). 

It would be cleaner to have an object like this 

```
{
     assignAuthor: true, 
     assignCategory: true
}
```

Reply:
This would break capability names for taxonomies containing hyphens.

So there's `wp:action-assign-wp_story_tag` but there could also be `wp:action-assign-super-awesome-tax`

Using `snakeCaseToCamelCase`  would turn this into something like `assignSuperAwesomeTax` which seems very unexpected and fragile. Because it means you need to use `snakeCaseToCamelCase` everywhere you'd want to check for that cap.